WebDriverAgentLib/FBApplication.m

```diff
@@ -35,21 +35,44 @@ @interface FBApplication ()
 @implementation FBApplication
 
 + (instancetype)fb_activeApplication
+{
+  return [self fb_activeApplicationWithDefaultBundleId:nil];
+}
+
++ (instancetype)fb_activeApplicationWithDefaultBundleId:(nullable NSString *)bundleId
 {
   NSArray<XCAccessibilityElement *> *activeApplicationElements = [FBXCAXClientProxy.sharedClient activeApplications];
-  XCAccessibilityElement *activeApplicationElement = [activeApplicationElements firstObject];
+  XCAccessibilityElement *activeApplicationElement = nil;
   if (activeApplicationElements.count > 1) {
-    XCAccessibilityElement *currentElement = self.class.fb_onScreenElement;
-    if (nil != currentElement) {
-      for (XCAccessibilityElement *appElement in activeApplicationElements) {
-        if (appElement.processIdentifier == currentElement.processIdentifier) {
-          activeApplicationElement = appElement;
+    if (nil != bundleId) {
+      // Try to select the desired application first
+      NSArray<NSDictionary *> *appsInfo = [self fb_appsInfoWithAxElements:activeApplicationElements];
+      NSUInteger index = 0;
+      for (NSDictionary *appInfo in appsInfo) {
+        index++;
```

Comment:
I'm a bit confused here. When we are iterating over the appsInfo and we find an appInfo where bundleIds are equal, we will then get index + 1 object in the activeApplicationElements. Is this intended?

Reply:
good catch, we need to increment the index after the condition